### PR TITLE
Add CookieBot compatibility.

### PR DESCRIPTION
**Pull request by chris-pook**

### CODE REVIEW

**Comment by dnlowman** [comment]
Would it be possible to test this change on one of our Magento test stores? Not sure if we have Cookiebot ourselves to test.

File: app/code/Ometria/Core/Helper/Cookiechannel.php [hunk 1]
```diff
@@ -1,97 +1,185 @@
...
+                    $cookieAllowed = true;
+                    break;
+
+                default:
+                    // The user must give their consent
+                    $cookieConsent = $this->getCookiebotConsent();
+                    if ($cookieConsent) {
+                        $cookieClass = $this->helperConfig->getCookiebotClass();
+                        // Consent cookie was found
+                        if (isset($cookieConsent[$cookieClass]) && filter_var($cookieConsent[$cookieClass], FILTER_VALIDATE_BOOLEAN)) {
```
Comment (weigruf) [inline app/code/Ometria/Core/Helper/Cookiechannel.php:153]: Do we need to check if the key exists as a first check?

`if (array_key_exists($cookieClass, $cookieConsent) && ...`

File: app/code/Ometria/Core/Helper/Cookiechannel.php [hunk 2]
```diff
@@ -1,97 +1,185 @@
...
 
-        //$ometria_config_helper = Mage::helper('ometria/config');
-        $ometria_config_helper = $this->helperConfig;
-        if (!$ometria_config_helper->isConfigured()) return;
-        if (!$ometria_config_helper->isUnivarEnabled()) return;
+        if (!$this->helperConfig->isUnivarEnabled()) {
+            return;
+        }
+
+        if ($command[0] == 'identify') {
```
Comment (weigruf) [inline app/code/Ometria/Core/Helper/Cookiechannel.php:63]: I'd say better to use a strict comparison here

File: app/code/Ometria/Core/Helper/Cookiechannel.php (hunk 1 above)
Comment (weigruf) [reply app/code/Ometria/Core/Helper/Cookiechannel.php:153]: Actually, as you have wrapped the $cookieConsent in a truthy check should be ok

**Review by weigruf — APPROVED** [review]

File: app/code/Ometria/Core/Helper/Cookiechannel.php (hunk 2 above)
Comment (chris-pook) [reply app/code/Ometria/Core/Helper/Cookiechannel.php:63]: I agree. Most of the changes here are refactoring the existing legacy code for better readability. In general we do this whenever extending the extension in an existing file with new functionality and aim to change the code as little as possible for obvious reasons. The legacy code from the initial build of the extension is quite messy so this approach is intended to improve things over time with minimal risk of adding breaking changes.

I will switch this to === now though, as it is the correct comparison to make.

**Pull request merged**